client/app/scripts/utils/router-utils.js

```diff
@@ -3,12 +3,31 @@ import page from 'page';
 import { route } from '../actions/app-actions';
 import AppStore from '../stores/app-store';
 
+//
+// TODO: move this logic somewhere else.
+//
+function shouldReplaceState(prevState, nextState) {
+  return (
```

Comment:
Yes, yes it is. Very good. https://www.wolframalpha.com/input/?i=%28P+and+Q%29+or+%28P+and+not+Q%29

I like the venn diagram.

I do like the expanded form as it does reveal the idea behind it. Which is these particular transitions, (Term -> Term or Term -> No Term). I'll give them `const` names though to make that clearer.